micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -776,6 +778,22 @@ public Config commonTags(String... tags) {
          * @return This configuration instance.
          */
         public synchronized Config meterFilter(MeterFilter filter) {
+            if (!meterMap.isEmpty()) {
+                InternalLogger logger = InternalLoggerFactory.getInstance(MeterRegistry.this.getClass());
+                String baseMessage = "A MeterFilter is being configured after a Meter has been registered to this registry. This is unexpected and should be corrected so all MeterFilters are configured before any Meters are registered. If that is not possible or you have a use case where it should be allowed, contact the Micrometer maintainers.";
+                if (logger.isDebugEnabled()) {
+                    String stackTrace = Arrays.stream(Thread.currentThread().getStackTrace())
+                        // skip getStackTrace frame and the call within this method
+                        .skip(2)
```

Comment:
How brittle is this? Can this bite us later?

Reply:
The JavaDoc has the following warning:

> Some virtual machines may, under some circumstances, omit one or more stack frames from the stack trace. In the extreme case, a virtual machine that has no stack trace information concerning this thread is permitted to return a zero-length array from this method.

For what it's worth, I did try with a few different JVM versions/vendors and skip(2) worked fine, but that's not a great guarantee. It does feel like the first two frames should be relatively stable since it is the getStackTrace call and the call to the containing method. If there were user code in there, it would be more risky.

That all said, it's safest to just not skip anything at the expense of the stack trace being more verbose with a couple frames we don't need. I'll update to remove the skipping.

Reply:
Without skipping (and I changed the test code a bit), we end up with a stack trace like this (with IntelliJ collapsing a bunch of reflection/junit/intellij frames):

```
16:01:44.236 [main] DEBUG i.m.c.i.simple.SimpleMeterRegistry - A MeterFilter is being configured after a Meter has been registered to this registry. This is unexpected and should be corrected so all MeterFilters are configured before any Meters are registered. If that is not possible or you have a use case where it should be allowed, contact the Micrometer maintainers.
java.base/java.lang.Thread.getStackTrace(Thread.java:1602)
	at io.micrometer.core.instrument.MeterRegistry$Config.meterFilter(MeterRegistry.java:785)
	at io.micrometer.core.instrument.MeterRegistry$Config.commonTags(MeterRegistry.java:759)
	at io.micrometer.core.instrument.MeterRegistry$Config.commonTags(MeterRegistry.java:771)
	at io.micrometer.core.instrument.MeterRegistryLoggingTest.configureCommonTags(MeterRegistryLoggingTest.java:79)
	at io.micrometer.core.instrument.MeterRegistryLoggingTest.registerMetricsAndConfigure(MeterRegistryLoggingTest.java:75)
	at io.micrometer.core.instrument.MeterRegistryLoggingTest.meterRegistrationBeforeMeterFilterConfigWithDebugLogging(MeterRegistryLoggingTest.java:57) <31 internal lines>
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541) <9 internal lines>
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541) <28 internal lines>
```

The first two frames would have been skipped before.